tests/ui/float_iterator_hint.rs

```diff
@@ -0,0 +1,15 @@
+// #106728
```

Comment:
New tests shouldn't be added to `tests/ui` - `tests/ui/iterators/integral.rs` might be more appropriate